builder/chroot-script.sh

```diff
@@ -218,38 +140,35 @@ ln -s /boot/meta-data /var/lib/cloud/seed/nocloud-net/meta-data
 # Fix duplicate IP address for eth0, remove file from os-rootfs
 rm -f /etc/network/interfaces.d/eth0
 
-# install docker-machine
-curl -sSL -o /usr/local/bin/docker-machine "https://github.com/docker/machine/releases/download/v${DOCKER_MACHINE_VERSION}/docker-machine-Linux-armhf"
-chmod +x /usr/local/bin/docker-machine
-
-# install bash completion for Docker Machine
-curl -sSL "https://raw.githubusercontent.com/docker/machine/v${DOCKER_MACHINE_VERSION}/contrib/completion/bash/docker-machine.bash" -o /etc/bash_completion.d/docker-machine
-
-# install docker-compose
-apt-get install -y \
-  --no-install-recommends \
-  python
-curl -sSL https://bootstrap.pypa.io/get-pip.py | python
-pip install "docker-compose==${DOCKER_COMPOSE_VERSION}"
-
-# install bash completion for Docker Compose
-curl -sSL "https://raw.githubusercontent.com/docker/compose/${DOCKER_COMPOSE_VERSION}/contrib/completion/bash/docker-compose" -o /etc/bash_completion.d/docker-compose
-
 # install docker-ce (w/ install-recommends)
-apt-get install -y --allow-downgrades \
+apt-get install --assume-yes --allow-downgrades \
   --no-install-recommends \
   "docker-ce=${DOCKER_CE_VERSION}"
 
-# install bash completion for Docker CLI
-curl -sSL https://raw.githubusercontent.com/docker/docker-ce/master/components/cli/contrib/completion/bash/docker -o /etc/bash_completion.d/docker
+# install bash completion
+declare -A downloads=(
+    ["/etc/bash_completion.d/docker-machine"]="https://raw.githubusercontent.com/docker/machine/v${DOCKER_MACHINE_VERSION}/contrib/completion/bash/docker-machine.bash"
+    ["/etc/bash_completion.d/docker-compose"]="https://raw.githubusercontent.com/docker/compose/${DOCKER_COMPOSE_VERSION}/contrib/completion/bash/docker-compose"
+    ["/etc/bash_completion.d/docker"]="https://raw.githubusercontent.com/docker/docker-ce/master/components/cli/contrib/completion/bash/docker"
+    ["/usr/local/bin/docker-machine"]="https://github.com/docker/machine/releases/download/v${DOCKER_MACHINE_VERSION}/docker-machine-Linux-armhf"
+    ["/usr/local/bin/rpi-serial-console"]="https://raw.githubusercontent.com/lurch/rpi-serial-console/master/rpi-serial-console"
+)
+for file in "${!downloads[@]}"; do
+    curl -fsSL "${downloads[$file]}" -o "$file"
+done
+chmod +x /usr/local/bin/docker-machine
+chmod +x /usr/local/bin/rpi-serial-console
 
-echo "Installing rpi-serial-console script"
-wget -q https://raw.githubusercontent.com/lurch/rpi-serial-console/master/rpi-serial-console -O usr/local/bin/rpi-serial-console
-chmod +x usr/local/bin/rpi-serial-console
+python -m pip install "docker-compose==${DOCKER_COMPOSE_VERSION}"
 
 # fix eth0 interface name
 ln -s /dev/null /etc/systemd/network/99-default.link
 
+# Neither are needed. Both can be fairly simply be installed using cloud-init or "apt-get update && apt-get install git/wget"
```

Comment:
Sure but then they sould not get installed in the first place 
https://github.com/hypriot/os-rootfs/blob/master/builder/build.sh#L27
But the idea behind hypriotos is that it is easy to use for everyone and wget and git are needed for many RPI projects to setup.

Reply:
That's fair. I will undo this one then.